tgui/packages/tgui/interfaces/common/sorting/SortableTable/type.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { BooleanLike } from "common/react";
+import { InfernoNode } from "inferno";
+import { SortDirection } from "../type";
+
+/** Configuration for Sortable Table header row and callback functions */
+export interface SortableTableHeaderConfig {
+
+  /** Presentation of header cell for this field */
+  children: InfernoNode,
+
+  /** Is this field sortable? */
+  sortable: BooleanLike,
+
+  /** Is this field searchable */
+  searchable: BooleanLike,
+
+  /** Optional tooltip string. */
+  toolTipContent?: string,
+
+  /** Callback function to sort on type specific to data given */
+  compareFunc?: (a: unknown, b: unknown) => number,
+
+  /** Conversion from your data to string to allow searching */
+  toString?: (data: unknown) => string,
+
+}
+
+export type SortState = {
+  /** Index into SortableTableHeaderConfig */
+  index: number,
```

Comment:
Prefer a more descriptive term here, `columnIndex` (I think that's what this is?)